src/extension/provider/launcher.ts

```diff
@@ -58,10 +59,16 @@ export class RunmeFile extends TreeItem {
 export class RunmeLauncherProvider implements TreeDataProvider<RunmeFile>, Disposable {
   #disposables: Disposable[] = []
 
+  protected runner?: IRunner
+
   private filesTree: Map<string, TreeFile> = new Map()
   private defaultItemState: TreeItemCollapsibleState = TreeItemCollapsibleState.Collapsed
 
-  constructor(private workspaceRoot?: string | undefined) {
+  constructor(
+    runner: IRunner | undefined,
```

Comment:
I think there is never a case where we don't provide the runner, right?
```suggestion
    protected runner: IRunner,
```

Reply:
I was trying to follow this example here.
https://github.com/stateful/vscode-runme/blob/b242c5e71a39d8cd4063c2f7a20ce2eb9fa9deca/src/extension/extension.ts#L62